File: Java/javaperformancetester/src/main/java/edu/imepac/javaperformancetester/controllers/VagaoController.java
package edu.imepac.javaperformancetester.controllers;

import edu.imepac.javaperformancetester.models.Vagao;
import edu.imepac.javaperformancetester.services.VagaoService;
import io.micrometer.core.annotation.Timed;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/vagao")
public class VagaoController {

    @Autowired
    private VagaoService vagaoService;

    @PostMapping
    public ResponseEntity<Vagao> criar(@RequestBody Vagao vagao){
        Vagao salvo = vagaoService.create(vagao);
        return ResponseEntity.status(HttpStatus.CREATED).body(salvo);
    }

    @Timed(value = "api.vagao.insercao", description = "Tempo de inserção do vagao")
    @PostMapping("/salvaremlote")
    public ResponseEntity<List<Vagao>> salvartodos(@RequestBody List<Vagao> vagoes){
        List<Vagao> salvos = vagaoService.salvarTodos(vagoes);
        return ResponseEntity.status(HttpStatus.CREATED).body(salvos);
    }

    @Timed(value = "api.vagao.listagem", description = "Tempo de listagem do vagao")
    @GetMapping
    public ResponseEntity<List<Vagao>> listarTodos(){
        List<Vagao> vagoes = vagaoService.findAll();
        return ResponseEntity.status(HttpStatus.OK).body(vagoes);
    }
}

File: Kotlin/kotlinperformancetester/src/main/kotlin/edu/imepac/kotlinperformancetester/controllers/VagaoController.kt
package edu.imepac.kotlinperformancetester.controllers

import edu.imepac.kotlinperformancetester.models.Vagao
import edu.imepac.kotlinperformancetester.services.VagaoService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/vagao")
class VagaoController {
    @Autowired
    private lateinit var vagaoService: VagaoService

    @PostMapping
    fun criar(@RequestBody vagao: Vagao): ResponseEntity<Vagao> {
        val vagaoSalvo = vagaoService.create(vagao)
        return ResponseEntity.status(HttpStatus.CREATED).body(vagaoSalvo)
    }

    @PostMapping("/salvaremlote")
    fun salvarTodos(@RequestBody vagoes: List<Vagao>): ResponseEntity<List<Vagao>> {
        vagaoService.salvarTodos(vagoes)
        return ResponseEntity.status(HttpStatus.CREATED).body(vagoes)
    }

    @GetMapping
    fun listarTodos(): ResponseEntity<List<Vagao>> {
        val vagoes = vagaoService.findAll()
        return ResponseEntity.status(HttpStatus.OK).body(vagoes)
    }

}